File: app/src/main/java/com/cham/helx/madapter/Viewpager2FragmentAdapter.java
package com.cham.helx.madapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.List;

/**
 * Hello World
 * Date: 2019/10/19
 * Author: Cham
 */
public class Viewpager2FragmentAdapter  extends FragmentStateAdapter {


    private List<Fragment> mFragmentList;


    public Viewpager2FragmentAdapter(@NonNull FragmentActivity fragmentActivity,List<Fragment> fragmentList) {
        super(fragmentActivity);
        mFragmentList =fragmentList;

    }

    public Viewpager2FragmentAdapter(@NonNull Fragment fragment) {
        super(fragment);
    }

    public Viewpager2FragmentAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getItemCount() {
        return mFragmentList.size();
    }
}

File: alice/src/main/java/com/cham/alice/base/Viewpager2FragmentAdapter.kt
package com.cham.alice.base

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

/**
 * Hello World
 * Date: 2019/11/11
 * Author: Cham
 */
class Viewpager2FragmentAdapter(var fragmentActivity: FragmentActivity,var fragmentList:MutableList<Fragment>) :
        FragmentStateAdapter(fragmentActivity) {


    override fun getItemCount(): Int {
     return  fragmentList.size
    }

    override fun createFragment(position: Int): Fragment {
        return  fragmentList[position]
    }

    override fun containsItem(itemId: Long): Boolean {
        return super.containsItem(itemId)
    }


    override fun getItemId(position: Int): Long {
        return super.getItemId(position)
    }

}